### PR TITLE
5sec camera

### PR DESCRIPTION
**Pull request by hanshengzhao1993**

### CODE REVIEW

File: src/public/components/CameraPage.js [hunk 1]
```diff
@@ -42,13 +43,28 @@ export default class CameraPage extends React.Component {
...
+  startCamera () {
+    // testing purposes making it so its only 1 minute after cut off time
+    // put in however much time you need for how much time afterwards
+    let end = Moment(this.state.selectedTimeCutoff).add(1,'minute');
+    let startCam = setInterval( ()=> {
+      let currentTime = Moment();
+      //uncomment this if you are testing the automated camera
+      this.takeScreenshot();
+      if ( currentTime.isAfter(end) ) {
+        console.log('stopping the interval 1 minute after cutoff time')
```
Comment (Jukejc) [inline src/public/components/CameraPage.js]: console.log

File: src/public/components/CameraPage.js [hunk 2]
```diff
@@ -42,13 +43,28 @@ export default class CameraPage extends React.Component {
...
+        //stop taking pictures of the camera
+        clearInterval(startCam)
+      };
+      console.log('still counting in setInteval')
```
Comment (Jukejc) [inline src/public/components/CameraPage.js]: console.log

File: src/server/db/ModelHelpers/attendanceHelpers.js [hunk 3]
```diff
@@ -22,14 +22,31 @@ exports.storeRecords = async (req, res) => {
...
         clearInterval(absentInterval);
       };
     }, 5000);
 
+    // this is from 9:00-9:30am
+    var tardyInterval = setInterval( ()=>{
+      let currentTime = moment();
+      let tardyEmail = moment(time).add(10, 'minute');
+      if( currentTime.isAfter(tardyEmail)) {
+        console.log('coming into the third setInterval and sending late emails')
```
Comment (Jukejc) [inline src/server/db/ModelHelpers/attendanceHelpers.js]: console.log

File: src/server/db/ModelHelpers/attendanceHelpers.js [hunk 4]
```diff
@@ -22,14 +22,31 @@ exports.storeRecords = async (req, res) => {
...
       if( currentTime.isAfter(time) ) {
-        Attendance.emailLateStudents();
+        // still send late emails 
+        // leave all late students to be pending;
+       Attendance.emailStudentAboutToBeTardy();
         clearInterval(absentInterval);
       };
     }, 5000);
 
+    // this is from 9:00-9:30am
```
Comment (andrewalonis) [inline src/server/db/ModelHelpers/attendanceHelpers.js]: should be something like: 'this is from cutoff time to 30 minutes afterwards'

File: src/server/db/QueryModels/AttendanceModel.js [hunk 5]
```diff
@@ -45,16 +45,30 @@ export default class AttendanceModel extends AttendanceQueries {
...
+    const [users] = await db.queryAsync(getAllPendingUsersEmails);
+    await sendTardyEmails(users);
+  }
+
   async emailLateStudents() {
-    const getLateUsersQuery = super.getPendingUsers();
-    const getAllLateEmailsQuery = super.getAllLateUserEmails();
-    const [users] = await db.queryAsync(getLateUsersQuery);
+    const getPendingUsersQuery = super.getPendingUsers();
+    // const getAllPendingUsersEmails = super.getAllPendingUsersEmails();
```
Comment (andrewalonis) [inline src/server/db/QueryModels/AttendanceModel.js]: Is this still necessary?

Comment (Jukejc) [inline src/server/db/QueryModels/AttendanceModel.js]: is this commented out code being used?

File: src/server/db/QueryModels/AttendanceModel.js [hunk 6]
```diff
@@ -45,16 +45,30 @@ export default class AttendanceModel extends AttendanceQueries {
...
       await db.queryAsync(lateQuery);
     });
-    const [emails] = await db.queryAsync(getAllLateEmailsQuery);
-    await sendAbsentEmails(emails);
+
+    const getLateStudentsEmails = super.getAllLateUserEmails();
+    const [lateUsers] = await db.queryAsync(getLateStudentsEmails)
+    await sendAbsentEmails(lateUsers)
+
+    console.log(' this should be all the late users', lateUsers);
```
Comment (Jukejc) [inline src/server/db/QueryModels/AttendanceModel.js]: console.log

File: src/server/mailgun/mailGunHelpers.js [hunk 7]
```diff
@@ -23,6 +23,7 @@ exports.sendMailLate = (req, res) => {
 };
   
 exports.sendMailForArrival = async (matchedUsers) => {
+  console.log('sending email for arrival')
```
Comment (Jukejc) [inline src/server/mailgun/mailGunHelpers.js]: console.log

File: src/server/mailgun/mailGunHelpers.js [hunk 8]
```diff
@@ -35,7 +36,9 @@ exports.sendMailForArrival = async (matchedUsers) => {
 };
 
 exports.sendAbsentEmails = async (emails) => {
+  console.log('sending absent emails')
```
Comment (Jukejc) [inline src/server/mailgun/mailGunHelpers.js]: console.log

File: src/server/mailgun/mailGunHelpers.js [hunk 9]
```diff
@@ -35,7 +36,9 @@ exports.sendMailForArrival = async (matchedUsers) => {
...
   emails.forEach((user) => {
+    console.log(user)
```
Comment (Jukejc) [inline src/server/mailgun/mailGunHelpers.js]: console.log

File: src/server/mailgun/mailGunHelpers.js [hunk 10]
```diff
@@ -47,6 +50,7 @@ exports.sendAbsentEmails = async (emails) => {
 };
 
 exports.sendWarningEmails = async (emails) => {
+  console.log('sending warning emails')
```
Comment (Jukejc) [inline src/server/mailgun/mailGunHelpers.js]: console.log

File: src/server/mailgun/mailGunHelpers.js [hunk 11]
```diff
@@ -56,4 +60,17 @@ exports.sendWarningEmails = async (emails) => {
     };
     // mailgun.messages().send(data);
   });
-};
\ No newline at end of file
+};
+
+exports.sendTardyEmails = async (users) => {
+  console.log('sending tardy emails')
```
Comment (Jukejc) [inline src/server/mailgun/mailGunHelpers.js]: console.log

**Pull request merged**